### ISSUE TITLE
Remove dependence on obs_lsstSim for electronics readout

### ISSUE THREAD

**Issue by cwwalter** [LSSTDESC/imSim#121]
Remove dependence on obs_lsstSim for geometry and pupil to focal plane mapping.

Original issue below:
Currently sims_GalSimInterface still relies on obs_lsstSim for (only?) focal plane geometry information.  As that package is deprecated in favor of obs_lsstCam we should modify the code to take the information from the camera object instead.

**Comment by cwwalter** [comment]
The mapping between pupil coordinates and focal plane coordinates may also be stored here.  @danielsf can you confirm what sort of this information is in obs_lsstSim?

**Comment by jchiang87** [comment]
Scott can correct me, but I think all of the dependence in `sims_coordUtils`.

**Comment by danielsf** [comment]
As I understand it, `obs_lsstSim` programmatically generates the zeroth mapping from pupil to focal plane coordinates by reading in the `segementation.txt` and `focalplanelayout.txt` files copied from PhoSim.  The Zernike polynomials layered on top of that zeroth order transformation are generated in `sims_utils` and `sims_coordUtils`.

Is that what you are asking, Chris?

**Comment by cwwalter** [comment]
I closed #118 which referred to x-talk and other instrumentation information that should also be moved to obs_lsstCam (which came from a copy of a PhoSim segmentation file).

**Comment by jchiang87** [comment]
I started work on the changes for #118 over the weekend.

**Comment by cwwalter** [comment]
Ah, do you want me to re-open it? Or is it OK to track it here?

**Comment by jchiang87** [comment]
We can track it here.

**Comment by cwwalter** [comment]
@danielsf said:

> As I understand it, obs_lsstSim programmatically generates the zeroth mapping from pupil to focal plane coordinates by reading in the segementation.txt and focalplanelayout.txt files copied from PhoSim. The Zernike polynomials layered on top of that zeroth order transformation are generated in sims_utils and sims_coordUtils.

Yes, I'm not positive where this code should live (suggestions welcome) but I don't think we want to rely on obs_lsstSim at all.  It is going to be deprecated.  So, that first order fit should happen somewhere else.   Also note (#122) for creating the grid of points to fit to for the Zernike overlay.

**Comment by cwwalter** [comment]
@jchiang87 I edited the original issue description and made a checkmark list since this now tracks a few things.  Please edit / or add to if you think it can be improved..

**Comment by cwwalter** [comment]
Note: the issue to update sensor properties themselves to current project values is #119.

**Comment by jchiang87** [comment]
Regarding the second item in the to-do list in the issue description:

Looking at `sims_coordUtils` and `sims_utils`, I think the only changes outside of `imSim` that need to be implemented are just replacing the references to `obs_lsstSim.LsstSimMapper` with corresponding references to `obs_lsstCam.LsstCamMapper` in https://github.com/lsst/sims_coordUtils/blob/master/python/lsst/sims/coordUtils/LsstCameraMethod.py
and fixing up the unit test code in `sims_coordUtils`, since those tests were written to test against the phosim configurations included in `obs_lsstSim`.   I **think** that all of the Zernike fitting should then be taken care of automatically when we use the various coordinate transformation utilities in `sims_coordUtils`.

**Comment by danielsf** [comment]
I agree with Jim.  Changing the `lsst_camera()` method in `sims_coordUtils` to instantiate the camer from `obs_lsstCam` will be enough to get imSim using `obs_lsstCam` for focal plane geometry.

Is `obs_lsstCam` stable enough to start shipping with `lsst_sims`?  If not, we might need to do some refactoring to make it possible for users to choose between `obs_lsstCam` and `obs_lsstSim` when using `sims_coordUtils`.

**Comment by cwwalter** [comment]
obs_lsstCam is planned to be distributed in lsst_distrib.  I just checked and it isn't there yet as of _31 but it should be soon (I think).

**Comment by cwwalter** [comment]
Are there any users other than imSim for this? I mean can we just cut over rather than allowing a choice?

**Comment by danielsf** [comment]
There are MAF metrics which use the camera model in `lsst_sims`.  I don't think they would care as long as there is a realistic model of chip gaps *somewhere* in `lsst_sims`.

(Do we actually expect the various coordinate transformations to change between `obs_lsstSim` and `obs_lsstCam`?)

**Comment by cwwalter** [comment]
I moved putting the new sensor parameters into its own issue (#130).

**Comment by cwwalter** [comment]
> (Do we actually expect the various coordinate transformations to change between obs_lsstSim and obs_lsstCam?)

Hmm... I don't *think* so...  As I understand it, the plan going forward is to deprecate obs_lsstSim.  So if there are items (like chip gaps) or anything else we need that is in obs_lsstSim that might not be in obs_lsst, we should figure out another place for them to go.

**Comment by jchiang87** [comment]
I think there will be changes of the kind Chris mentions, especially if we have a mixed focal plane in obs_lsstCam (which we will have in reality).  My understanding is that all of that sort of thing would in fact be in obs_lsstCam, i.e., the details of the focalplane geometry in physical units (i.e., mm) that are in obs_lsstSim will definitely change and those changes should be captured in obs_lsstCam.

**Comment by danielsf** [comment]
I just tried naively dropping `obs_lsstCam` in place of `obs_lsstSim`.  All of the unit tests failed.  Is once `obs_lsstCam` is ready to ship, is there going to be a formal writeup of all of the changes (looking at some of the outputs, it looks like what was once called `R:0,1 S:0,0` is now `R10_S00`, among other things), or should I just start hacking away at it now?

**Comment by cwwalter** [comment]
We are changing the scope of this issue for 2.0i.  

I have changed the title to "Remove dependence on obs_lsstSim to Remove dependence on obs_lsstSim for electronics readout"

It turns out that the needed changes to remove the obs_lsstSim dependence in sims_GalSimInterface are not as trivial as expected.  So, we will still make this change for all of the electronics readout, x-talk paraders etc.  This will mean that will will not need to use the segmentation.txt file any more for parameters and that those parameters will go into the same camera objects being used by DM for measurement as described in #130.  

However, we will wait to change the geometry dependence (which is completely different than the electronics readout) used by sims_GalSimInterface until we have enough time to check it carefully and evaluate unit tests that our currently failing after this change.  If the schedule for the 2.0i production moves we can re-evaluate this decision.

**Comment by cwwalter** [comment]
Since the obs_lsstCam PR is still open, there are some nominal values it is still using now?
